Updates related to filtering processed inventories

bl-young opened the pull request:
- New parameter filter_list in getInventory(), one or more named filters can be passed to filter any stored inventory.
- filter.yaml file added to describe available filters, and where necessary include filter parameters. Allows for consolidation of input files related to chemical omit lists (e.g. NEI_pollutant_omit_list.csv)
- New function seeAvailableInventoryFilters() to print out the available filters and descriptions in filter.yaml
- Update to handling of fields to reduce redundancy and enable data specific fields to be maintained in stored parquet (e.g. RCRAInfo NBR fields)
- Reorganizes function in global, moving some to validate.py and filter.py
- Updates RCRAInfo to maintain more data in processed inventories, which can then be removed when accessing those inventories by filtering
- Minor improvements to processing of RCRAInfo

WesIngwersen commented:
This is looking good. I would recommend moving filter.yaml up a level to be at the same level as the config.yaml
The levels in filter.yaml are a little confusing - it's hard to detect any clear pattern and there seems to be some entries strictly for metadata, and filter.py is similiar in that there is a general filter_inventory function but there are some filters that don't use it.  However I understand that this is a work in progress.

bl-young commented:
> This is looking good. I would recommend moving filter.yaml up a level to be at the same level as the config.yaml
> The levels in filter.yaml are a little confusing - it's hard to detect any clear pattern and there seems to be some entries strictly for metadata, and filter.py is similiar in that there is a general filter_inventory function but there are some filters that don't use it. However I understand that this is a work in progress.

@WesIngwersen I've addressed these issues (specifically https://github.com/USEPA/standardizedinventories/commit/19fb8e896e249b3a036c84b6fec30645e96e3c8a and https://github.com/USEPA/standardizedinventories/commit/4deadb5c9b14400583c9d11722ab703379a42644). On filter.yaml, my intention was to provide a description of each filter, as well as any parameters if necessary. So yes it is combined metadata and parameters. I am open to alternatives, but wanted to keep everything in one spot.

bl-young commented:
@WesIngwersen I'd like to pull this in and do a 0.9.9 release so that I can begin final testing for v1.0 release.

WesIngwersen commented:
Is `filter_for_LCI` intentionally left in as a separate parameter than `filter_list`?  Is that mainly to make it more backward compatible?

bl-young commented:
> Is `filter_for_LCI` intentionally left in as a separate parameter than `filter_list`? Is that mainly to make it more backward compatible?

correct - I didn't want to remove that parameter yet. I'm thinking its probably used in eLCI

WesIngwersen reviewed (commented):
The code is working and my cosmetic and renaming updates are not changing that, but something screwy is happening with the filters I believe. I ran multipe filters and they are each removing the same # of rows from the df for TRI.

```
tri = stewi.getInventory("TRI",2016)

tri_no_states = stewi.getInventory("TRI", 2016, filters=['US_States_only'])
len(tri)-len(tri_no_states)
#281

tri_f_LCI = stewi.getInventory("TRI", 2016, filters=['filter_for_LCI'])
len(tri)-len(tri_f_LCI)
#281
```

Could you check and also assign another ERG reviewer to test this since its a pretty major refactor and we might not have checked and compared filters well initially?
